src/runner.ts

```diff
@@ -160,6 +171,22 @@ export class CoverageParserRunner {
         process.on("error", (err) => { reject(err); });
     }
 
+    private async isCoverageReport(report: string): Promise<boolean> {
```

Comment:
The current method parses the entire report file, which is quite expensive. You should try other approaches that can only check "Coverage" as the root node. If you previously encountered issues with the SAX Parser in scenarios involving multiple files, please check whether listeners and similar components were properly cleaned up.